call-home/src/main.rs

```diff
@@ -55,9 +55,9 @@ async fn run() -> anyhow::Result<()> {
     let endpoint = args.endpoint;
     let namespace = digest(args.namespace);
 
-    let sleep_duration = get_call_home_frequency();
-    let encryption_dir = get_encryption_dir();
-    let key_filepath = get_key_filepath();
+    let sleep_duration = call_home_frequency();
+    let encryption_dir = encryption_dir();
+    let key_filepath = key_filepath();
```

Comment:
@niladrih are these variables (encryption_dir and key_filepath) being consumed?

Reply:
Yes, these are being consumed inside the loop further down inside that function.
They are declared there to read input from the environment variables once, and the variable is used in the loop.

2 of the variables are cloned inside the loop because we are using them inside a `tokio::task::spawn_blocking()`. The spawn_blocking() function accepts a closure which can have arguments which live at least as long as the scope of spawn_blocking() call. In this case, that scope is the loop iteration of the spawn_blocking() call. The compiler cannot guarantee that the variables will continue to live for the entire duration of the spawn_blocking() call, as the arguments are `String` type (owned type). By redeclaring them inside the loop iteration, we have guaranteed that even if the outer scope variables are moved, the loop scoped clones will continue to exist at least as long as the iteration exists, and the iteration with will wait for the spawn_blocking() call to return before it exits and moves on to the next iteration.

Reply:
You may notice that the `report` variable does not have this problem, because the report variable will always be generated in each iteration and then moved into the closure. If the report variable was outside the loop, it will be consumed by the first iteration of the loop, and no longer usable, then we'd have to clone it inside the loop scope.

We're not using the functions encryption_dir() and key_filepath() inside the loop because these read from environment variables, and the input values are intended to be consistent throughout the duration of the process. So, input must be read once.